dashboard/config/initializers/atomic_save.rb

```diff
@@ -0,0 +1,10 @@
+# Provides `#atomic_save!` which skips the default transaction
+# wrapping behavior of `#save!`.
+#
+# Only use in cases where validations/callbacks will never
+# need to trigger any rollback of the operation.
+module ActiveRecord
+  module Persistence
+    alias_method :atomic_save!, :save!
+  end
+end
```

Comment:
See also #15282 for some previous [discussion](https://github.com/code-dot-org/code-dot-org/pull/15282/files#r117862179) on this addition.